src/transformers/modeling_utils.py

```diff
@@ -1512,6 +1513,40 @@ def _autoset_attn_implementation(
             # If a config is passed with a preset attn_implementation, we skip the automatic dispatch and use the user-provided config, with hard checks that the requested attention implementation is available.
             requested_attn_implementation = config._attn_implementation_internal
 
+        # MultiModal LLM related hack: since they consist of two or might be more sub-models
+        # we have to check and dispatch SDPA to each sub-model, in case any of them support it.
+        # If one sub-model supports SDPA while other doesn't, an error will be raised following the
+        # typical SDPA-dispatch path.
+        # Same goes for the `_supports_cache_class` because we cannot know what flag LM has
+        # before knowing which class is that LM
+        if hasattr(config, "text_config") and hasattr(config, "vision_config"):
+            text_model_cls = MODEL_MAPPING.get(type(config.text_config), None)
+            if text_model_cls is not None:
+                config.text_config._attn_implementation = config._attn_implementation
+                cls._supports_sdpa = text_model_cls._supports_sdpa
+                cls._autoset_attn_implementation(
+                    config.text_config,
+                    use_flash_attention_2=use_flash_attention_2,
+                    torch_dtype=torch_dtype,
+                    device_map=device_map,
+                    check_device_map=check_device_map,
+                )
+
+            vision_model_cls = MODEL_MAPPING.get(type(config.vision_config), None)
+            if vision_model_cls is not None:
+                config.vision_config._attn_implementation = config._attn_implementation
+                cls._supports_sdpa = vision_model_cls._supports_sdpa
+                cls._autoset_attn_implementation(
+                    config.vision_config,
+                    use_flash_attention_2=use_flash_attention_2,
+                    torch_dtype=torch_dtype,
+                    device_map=device_map,
+                    check_device_map=check_device_map,
+                )
+
+            if vision_model_cls is not None and text_model_cls is not None:
+                cls._supports_sdpa = vision_model_cls._supports_sdpa or text_model_cls._supports_sdpa
```

Comment:
It's very bad to modify a class attribute: it will lead bugs and nightmares 😰   very difficult to detect.

One example is the `InstructBlip`, see #27867 and #25910.

I see you need it in the call below to `_check_and_enable_sdpa`. Got to think another way ...

Reply:
BTW, 

```
        # If one sub-model supports SDPA while other doesn't, an error will be raised following the
        # typical SDPA-dispatch path.
```

If I understand correctly, the call to `_check_and_enable_sdpa`  and `_autoset_attn_implementation` will return without error, right? The error occur only when we try to init. the model with this config?

Reply:
Yes, the error for SDPA is only when the user specifically sets `attn_implementation=sdpa` when loading the model. Dispatching SDPA on sub-configs of VLM will follow the same path so that we raise error that "VisionMode doesn't support sdpa" only if user specifically requested sdpa. Otherwise we dispath to eager by default